automated-record-card/index.html

```diff
@@ -0,0 +1,196 @@
+        <button id="reset-record" class="cf-btn-secondary">Reset</button>
+      </p>
+      <div id="message" class="cf-form-hint"></div>
+    </div>
+
+    <script>
+      'use strict';
+
+      window.contentfulExtension.init(extension => {
+        extension.window.startAutoResizer();
+
+        const inputRecordID = document.querySelector('.cf-form-input');
+        const fieldRecordID = extension.field;
+        let valueRecordID = fieldRecordID.getValue();
+        validateInput();
+
+        if (extension.entry.fields.thumbnailUrl.getValue()) {
+          document.getElementById('thumbnail-preview').innerHTML = '<img src="' + extension.entry.fields.thumbnailUrl.getValue() + '&size=w200" alt="" style="width: auto; height: 100%;" />'
+        }
+
+        const detachValueChangeHandler = fieldRecordID.onValueChanged(valueChangeHandler);
+        inputRecordID.addEventListener('input', keyboardInputHandler);
+        inputRecordID.addEventListener('blur', validateInput);
+        window.addEventListener('onbeforeunload', unloadHandler);
+
+        function validateInput() {
+          if (valueRecordID !== undefined && valueRecordID !== inputRecordID.value) {
+            document.getElementById('find-record').click();
+          }
+        }
+
+        function valueChangeHandler (value) {
+          inputRecordID.value = value || '';
+        }
+
+        function keyboardInputHandler () {
+          const value = inputRecordID.value;
+          if (typeof value !== 'string' || value === '') {
+            fieldRecordID.removeValue();
+          } else {
+            fieldRecordID.setValue(value);
+          }
+        }
+
+        function unloadHandler () {
+          window.removeEventListener('onbeforeunload', unloadHandler);
+          inputRecordID.removeEventListener('input', keyboardInputHandler);
+          detachValueChangeHandler();
+        }
+
+        // find a value in one of the defined languages
+        function languageValue (value) {
+          const languageKeys = ['en', 'eng', 'def', 'und'];
+          for (let key of languageKeys) {
+            if (value[key]) {
+              return value[key][0];
+            }
+          }
+          return;
+        }
+
+        // clear all fields
+        function clearEntry () {
+          // clear message
+          document.getElementById('message').innerText = '';
+
+          // clear fields
+          extension.entry.fields.name.removeValue();
+          extension.entry.fields.creator.removeValue();
+          extension.entry.fields.provider.removeValue();
+          extension.entry.fields.thumbnailUrl.removeValue();
+
+          document.getElementById('thumbnail-preview').innerHTML = '';
+        }
+
+        // click reset button
+        document.getElementById('reset-record').addEventListener('click', e => {
+          e.preventDefault();
+          clearEntry();
+          // clear identifier
+          extension.entry.fields.identifier.removeValue();
```

Comment:
This seems brittle. I suspect we will want some way of making this agnostic to the name of the identifier field.

Reply:
Removed it completely as it is already cleared with the clearEntry()